### PR TITLE
[SPARK-432] support: DEV=true make universe

### PR DESCRIPTION
**Pull request by mgummelt**
Previously, in order to build the DC/OS spark package from source, you would have to manually build spark, and then set SPARK_DIST_URI to the file:// location of the distribution.

This introduces the DEV=true env var, which when set, will do all this for you.  It required reverse engineering spark's make-distribution.sh script.

### CODE REVIEW

**Comment by mgummelt** [comment]
cc @susanxhuynh for review

**Comment by mgummelt** [comment]
retest this please

**Comment by susanxhuynh** [comment]
Why does it adapt spark's make-distribution.sh script, rather than calling it directly?

File: dispatcher/bin/universe.sh [hunk 1]
```diff
@@ -17,14 +19,20 @@ set -e -x -o pipefail
...
     if [ -z "${SPARK_DIST_URI}" ]; then
-        SPARK_DIST_URI=$(jq ".default_spark_dist.uri" "${SPARK_BUILD_DIR}/manifest.json")
-        SPARK_DIST_URI="${SPARK_DIST_URI%\"}"
-        SPARK_DIST_URI="${SPARK_DIST_URI#\"}"
-        echo "Using Spark dist URI: $SPARK_DIST_URI"
+        if [[ "${DEV}" = "true" ]]; then
+            (cd "${SPARK_BUILD_DIR}" && make dist)
+            SPARK_DIST_URI="file://${SPARK_DIR}/spark-SNAPSHOT.tgz"
+        else
+            SPARK_DIST_URI=$(default_spark_dist)
```
Comment (susanxhuynh) [inline dispatcher/bin/universe.sh]: Where is 'default_spark_dist' defined?

Comment (mgummelt) [inline dispatcher/bin/universe.sh]: jenkins.sh

**Comment by mgummelt** [comment]
make-distribution.sh does a clean mvn build.  Ours does an incremental sbt build, which is much, much faster.

**Comment by susanxhuynh** [comment]
OK. LGTM

**Comment by mgummelt** [comment]
@susanxhuynh approved?

**Pull request merged**